mlir/lib/IR/BuiltinTypes.cpp

```diff
@@ -242,6 +243,11 @@ LogicalResult VectorType::verify(function_ref<InFlightDiagnostic()> emitError,
     return emitError() << "number of dims must match, got "
                        << scalableDims.size() << " and " << shape.size();
 
+  if (layout) {
+    if (failed(layout.verifyLayout(shape, elementType, emitError)))
+      return emitError() << "layout does not match underlying vector shape";
```

Comment:
the error message should be more generic as the mismatch may be in the shape or somewhere else.

Reply:
Makes sense, will change to something more generic like "Layout verification failed!"